utilities/poap/puppet_agent_install.py

```diff
@@ -0,0 +1,239 @@
+# basic tftp ip checksum. The script will also perform an integrity check
+# on the image and configuration files by downloading corresponding files
+# with .md5 extensions. The md5sum value can be generated for this file by
+# issuing the following command:
+#  f=puppet_agent_install.py ; cat $f | sed '/^#md5sum/d' > $f.md5 ; sed -i "s/^#md5sum=.*/#md5sum=$(md5sum $f.md5 | sed 's/ .*//')/" $f
+#
+
+import os
+from datetime import datetime
+import subprocess
+import shlex
+import sys
+
+###############################################################################
+# START OF USER-CONFIGURABLE PARAMETERS
+# These user-configurable parameters should be updated to reflect the local
+# automation environment.
+###############################################################################
+
+# (Required) RPM_NAME = The puppet agent release RPM
+# **************************************************************
+# *** IMPORTANT: CHOOSE THE RIGHT RPM FOR YOUR ENVIRONMENT! ****
+# **************************************************************
+bash_shell_rpm = 'puppetlabs-release-pc1-cisco-wrlinux-5.noarch.rpm'
+guestshell_rpm = 'puppetlabs-release-pc1-el-7.noarch.rpm'
+RPM_NAME = bash_shell_rpm
+
+
+# (Required) RPM_URI = The download URI for the RPM
+# RPM_URI = 'ftp://1.2.3.4/'
+RPM_URI = 'http://yum.puppetlabs.com/'
+
+# (required) PUPPET_SERVER = The DNS name or IP address of the agent's puppet server
+PUPPET_SERVER = 'puppet-cvh.cisco.com'
+
+# (Optional) VRF = The agent's VRF to use for the RPM install
+VRF = 'management'
+
+# (Optional) HTTP_PROXY = local http proxy server
+# (Optional) HTTPS_PROXY = local https proxy server
+# (Optional) NO_PROXY = proxy exclusions
+HTTP_PROXY = 'http://proxy.esl.cisco.com:8080'
+HTTPS_PROXY = 'https://proxy.esl.cisco.com:8080'
+NO_PROXY = ''
+
+# (optional) DOMAIN = The domain name to use with the agent node
+DOMAIN = 'cisco.com'
+
+# (Optional) DNS = The DNS configuration to use for /etc/resolv.conf
+# Use triple-quote syntax for multiple lines:
+#  DNS = '''\n
+#  nameserver 1.2.3.4
+#  domain cisco.com
+#  '''
+DNS = '''\n
+nameserver 64.102.6.247
+nameserver 72.163.131.10
+nameserver 173.36.131.10
+search cisco.com
+domain cisco.com
+'''
+
+###############################################################################
+# End OF USER-CONFIGURABLE PARAMETERS
+# Do not modify the following section.
+###############################################################################
+
+# Script initializations. Note: os.environ vars always win over script vars
+PUPPET_BINARY = '/opt/puppetlabs/puppet/bin/puppet'
+PUPPET_CONFIGPRINT = 'sudo ' + PUPPET_BINARY + ' agent --configprint config'
+
+# Check for required vars
+reqd_vars = ['RPM_NAME', 'RPM_URI', 'PUPPET_SERVER']
+for v in reqd_vars:
+    if not (v in globals() and len(v)):
+        raise ValueError("Required parameter '%s' is not set" % v)
+
+# Build PREPEND_CMD to specify both sudo and optional vrf (namespace)
+PREPEND_CMD = 'sudo'
```

Comment:
It makes the code cleaner than adding a bunch of vrf checks throughout the code.